src/opentime/rationalTime.cpp

```diff
@@ -94,6 +94,69 @@ is_dropframe_rate(double rate)
     return std::find(b, e, rate) != e;
 }
 
+static bool
+parseFloat(char const* pCurr, char const* pEnd, bool allow_negative, double* result)
+{
+    if (pCurr >= pEnd || !pCurr) {
+        *result = 0.0;
+        return false;
+    }
+
+    double ret = 0.0;
+    double sign = 1.0;
+
+    if (*pCurr == '+')
+        ++pCurr;
+    else if (*pCurr == '-') {
+        if (!allow_negative) {
+            *result = 0.f;
+            return false;
+        }
+        sign = -1.0;
+        ++pCurr;
+    }
+
+    // get integer part
+    uint32_t uintPart = 0;
+    while (pCurr < pEnd) {
+        char c = *pCurr;
+        if (c < '0' || c > '9')
+            break;
+        uintPart = uintPart * 10 + c - '0';
+        ++pCurr;
+    }
+
+    ret = (double) uintPart;
+
+    // check for end of string or delimiter
+    if (pCurr == pEnd || *pCurr == '\0') {
+        *result = sign * ret;
+        return true;
+    }
+
+    // if the next character is not a decimal point, the string is malformed.
+    if (*pCurr != '.') {
+        *result = 0.f; // zero consistent with earlier error condition
+        return false;
+    }
+
+    ++pCurr; // skip decimal
+
+    double scaler = 0.1;
```

Comment:
On first read, I was stuck on whether this is a "scalar" or a scale factor. I see that it is the order of magnitude, or place-value scale. Maybe just a comment would help others avoid the confusion?

Reply:
I've renamed it "position_scale" for unambiguity.